File: arb-www/src/schema/rollingInsights/nba/teamProfile.ts
import { BaseDTO } from '../../BaseDTO';

/**
 * NBA Rolling Insights Team Profile
 */
export class NBARollingInsightsTeamProfile extends BaseDTO {
  team_id!: number;
  team!: string;
  abbrv!: string;
  arena!: string;
  mascot!: string;
  conf!: string;
  location!: string;
}

/**
 * Rolling Insights NBA Team Profile response wrapper
 */
export interface NBARollingInsightsTeamProfileResponse {
  league: string;
  data_type: string;
  data: {
    Nba: {
      TeamProfiles: NBARollingInsightsTeamProfile[];
    };
  };
  filtered_count: number;
  total_count: number;
}

File: arb-rs/src/schema/rolling_insights/nba/team_profile.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NBARollingInsightsTeamProfile {
    #[serde(rename = "team_id")]
    pub team_id: i32,
    pub team: String,
    pub abbrv: String,
    pub arena: String,
    pub mascot: String,
    pub conf: String,
    pub location: String,
}
